File: tuts/tut13/converging.pl
#!/usr/bin/perl

my @numbers;
while ($line = <STDIN>) {
	chomp $line;
	@nums = split / /, $line;
	for $num (@nums) {
		if ($num =~ /[0-9]/) {
			push @numbers, int($num);
		}
	}
}

$prev = shift(@numbers);
for $num (@numbers) {
	if (defined $diff) {
		if ($prev - $num > $diff) {
			print "not converging\n";
			exit(0);
		}
	}	

	$diff = $prev - $num;

	if ($prev <= $num) {
		print "not converging\n";
		exit(0);
	}

	$prev = $num;
}

print "converging\n";

File: tuts/tut13/converging.py
#!/usr/bin/python3

import sys

numbers = []
for line in sys.stdin:
    nums = line.split()
    for num in nums:
        numbers.append(int(num))

diff = numbers[0] - numbers[1] + 1
prev = numbers[0]
for num in numbers[1:]:
    if prev - num >= diff:
        print("not converging")
        exit(0)
    
    diff = prev - num

    if prev <= num:
        print("not converging")
        exit(0)

    prev = num


print("converging")
